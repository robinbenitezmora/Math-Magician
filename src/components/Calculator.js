import React, { useState } from 'react';
import calculate from '../logic/calculate';
// eslint-disable-next-line react/prefer-stateless-function
function Calculator() {
  const [calcState, setCalcState] = useState({ total: null, next: null, operation: null });
  const { total, next, operation } = calcState;
  const eventCalculator = (e) => {
    setCalcState((calcState) => calculate(calcState, e.target.textContent));
  };

  return (
    <div className="calculator">
      <div className="calculator-display">
        {(total || '') + (operation || '') + (next || '')}
      </div>
      <div className="calculator-buttons">
        <div className="calculator-buttons-row">
          <button type="button" onClick={eventCalculator} className="btn">AC</button>
          <button type="button" onClick={eventCalculator} className="btn">+/-</button>
          <button type="button" onClick={eventCalculator} className="btn">%</button>
          <button type="button" onClick={eventCalculator} className="btn-operation">รท</button>
        </div>
        <div className="calculator-buttons-row">
          <button type="button" onClick={eventCalculator} className="btn">7</button>
          <button type="button" onClick={eventCalculator} className="btn">8</button>
          <button type="button" onClick={eventCalculator} className="btn">9</button>
          <button type="button" onClick={eventCalculator} className="btn-operation">x</button>
        </div>
        <div className="calculator-buttons-row">
          <button type="button" onClick={eventCalculator} className="btn">4</button>
          <button type="button" onClick={eventCalculator} className="btn">5</button>
          <button type="button" onClick={eventCalculator} className="btn">6</button>
          <button type="button" onClick={eventCalculator} className="btn-operation">-</button>
        </div>
        <div className="calculator-buttons-row">
          <button type="button" onClick={eventCalculator} className="btn">1</button>
          <button type="button" onClick={eventCalculator} className="btn">2</button>
          <button type="button" onClick={eventCalculator} className="btn">3</button>
          <button type="button" onClick={eventCalculator} className="btn-operation">+</button>
        </div>
        <div className="calculator-buttons-row">
          <button type="button" onClick={eventCalculator} className="btn zero">0</button>
          <button type="button" onClick={eventCalculator} className="btn">.</button>
          <button type="button" onClick={eventCalculator} className="btn-operation">=</button>
        </div>
      </div>
    </div>
  );
}

export default Calculator;
