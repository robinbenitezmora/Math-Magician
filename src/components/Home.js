import React from 'react';

function Home() {
  return (
    <div className="home-page">
      <h2 className="home-title">Welcome to our page!</h2>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam
        voluptas, quae, quibusdam voluptates, voluptatibus quod voluptatum
        impedit quidem doloremque quos repellat. Quod, quibusdam! Quisquam
        voluptas, quae, quibusdam voluptates, voluptatibus quod voluptatum
        impedit quidem doloremque quos repellat. Quod, quibusdam!
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam
        voluptas, quae, quibusdam voluptates, voluptatibus quod voluptatum
        impedit quidem doloremque quos repellat. Quod, quibusdam! Quisquam
        voluptas, quae, quibusdam voluptates, voluptatibus quod voluptatum
        impedit quidem doloremque quos repellat. Quod, quibusdam!
      </p>
    </div>
  );
}

export default Home;
