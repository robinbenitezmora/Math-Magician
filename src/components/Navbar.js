import React from 'react';
import { NavLink } from 'react-router-dom';

function Navbar() {
  return (
    <div className="navbar">
      <h1 className="navbar-title">Math Magicians</h1>
      <div className="navbar-links">
        <NavLink to="/" className="links">Home</NavLink>
        <NavLink to="/Calculator" className="links">Calculator</NavLink>
        <NavLink to="/Quote" className="links">Quote</NavLink>
      </div>
    </div>
  );
}

export default Navbar;
