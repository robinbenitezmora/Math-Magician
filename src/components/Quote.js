import React from 'react';

function Quote() {
  return (
    <div className="quote">
      <h2 className="quote-title">Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. â€“William Paul Thurston</h2>
    </div>
  );
}

export default Quote;
