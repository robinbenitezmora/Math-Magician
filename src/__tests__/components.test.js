import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';
import Quote from '../components/Quote';
import Navbar from '../components/Navbar';
import Home from '../components/Home';

describe('Test Calculator component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Home page renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Quote page renders correctly', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Navbar renders correctly', () => {
    const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Simulate click event', () => {
    render(<Calculator />);
    const key4 = screen.getByText('4');
    const key5 = screen.getByText('5');
    const key6 = screen.getByText('6');
    fireEvent.click(key4);
    fireEvent.click(key5);
    fireEvent.click(key6);
    expect(screen.getByText('456'));
  });
});
