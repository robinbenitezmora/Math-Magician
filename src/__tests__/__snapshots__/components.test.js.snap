// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Calculator component Home page renders correctly 1`] = `
<div
  className="home-page"
>
  <h2
    className="home-title"
  >
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam voluptas, quae, quibusdam voluptates, voluptatibus quod voluptatum impedit quidem doloremque quos repellat. Quod, quibusdam! Quisquam voluptas, quae, quibusdam voluptates, voluptatibus quod voluptatum impedit quidem doloremque quos repellat. Quod, quibusdam!
  </p>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam voluptas, quae, quibusdam voluptates, voluptatibus quod voluptatum impedit quidem doloremque quos repellat. Quod, quibusdam! Quisquam voluptas, quae, quibusdam voluptates, voluptatibus quod voluptatum impedit quidem doloremque quos repellat. Quod, quibusdam!
  </p>
</div>
`;

exports[`Test Calculator component Navbar renders correctly 1`] = `
<div
  className="navbar"
>
  <h1
    className="navbar-title"
  >
    Math Magicians
  </h1>
  <div
    className="navbar-links"
  >
    <a
      aria-current="page"
      className="links active"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="links"
      href="/Calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      className="links"
      href="/Quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </div>
</div>
`;

exports[`Test Calculator component Quote page renders correctly 1`] = `
<div
  className="quote"
>
  <p
    className="quote-title"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. –William Paul Thurston
  </p>
</div>
`;

exports[`Test Calculator component renders correctly 1`] = `
<div
  className="calculator-page"
>
  <h2
    className="calculator-title"
  >
    Let's do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="calculator-display"
    />
    <div
      className="calculator-buttons"
    >
      <div
        className="calculator-buttons-row"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="btn-operation"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
      </div>
      <div
        className="calculator-buttons-row"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="btn-operation"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="calculator-buttons-row"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="btn-operation"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="calculator-buttons-row"
      >
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="btn-operation"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="calculator-buttons-row"
      >
        <button
          className="btn zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="btn"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="btn-operation"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
